src/js/libcs/RenderBackends.js

```diff
@@ -851,26 +851,36 @@ function exportWith(Context, wnd) {
             var blob = csctx.toBlob();
             releaseExportedObject();
             exportedCanvasURL = window.URL.createObjectURL(blob);
-            wnd.location.href = exportedCanvasURL;
+
+            downloadHelper(exportedCanvasURL);
+            window.URL.revokeObjectURL(exportedCanvasURL);
         } finally {
             csctx = origctx;
         }
     });
 }
 
 globalInstance.exportSVG = function() {
-    exportWith(SvgWriterContext, window.open('about:blank', '_blank'));
+    exportWith(SvgWriterContext);
 };
 
 globalInstance.exportPDF = function() {
-    var wnd = window.open('about:blank', '_blank');
     CindyJS.loadScript('pako', 'pako.min.js', function() {
-        exportWith(PdfWriterContext, wnd);
+        exportWith(PdfWriterContext);
     });
 };
 
 globalInstance.exportPNG = function() {
-    var data = csctx.canvas.toDataURL();
-    var wnd = window.open('about:blank', '_blank');
-    wnd.location.href = data;
+    downloadHelper(csctx.canvas.toDataURL());
+};
+
+
+var downloadHelper = function(data) {
+    var a = document.createElement("a");
+    document.body.appendChild(a);
+    a.style = "display: none";
+    a.href = data;
+    a.download = "CindyJSExport";
+    a.click();
+    document.body.removeChild(a);
```

Comment:
Ah, so instead of *showing* the exported content in a *new* tab, we now *download* it in the *current* tab. I missed that aspect (no new tab) when reading the comment by @kranich. Hmm… Probably makes sense for most practical use cases, but will be annoying when developing export functionality, where I want to have a quick look at the generated file without having to download and open it every time. Well, I'll find a way around when I gets to this.